components/sup/src/manager/service/mod.rs

```diff
@@ -183,6 +193,51 @@ impl Service {
         })
     }
 
+    pub fn load(spec: ServiceSpec,
+                gossip_listen: &GossipListenAddr,
+                http_listen: &http_gateway::ListenAddr)
+                -> Result<Service> {
+        let mut ui = UI::default();
+        let package = match PackageInstall::load(&spec.ident, Some(&Path::new(&*FS_ROOT_PATH))) {
+            Ok(package) => {
+                match spec.update_strategy {
+                    UpdateStrategy::AtOnce | UpdateStrategy::Rolling => {
+                        try!(util::pkg::maybe_install_newer(&mut ui, &spec, package))
```

Comment:
We actually shouldn't perform a `maybe_install_newer()` action here for `UpdateStrategy::Rolling`. We were before, but that was bad behaviour. When the service joins the group it's going to look at it's peers and see what version it should be on.

In the behaviour we had (and by proxy, this behaviour) a service with a broad goal of a package ident could be started, update, and then join a service group which is in the rolling strategy. As soon as it joins it could notice it's peers are on a different version - maybe the election leader had a more strict goal perhaps - and then the service would downgrade itself and *then* join the service group.

tldr; add `UpdateStrategy::Rolling` to the same match arm as `UpdateStrategy::None` :)